src/js/users/types.ts

```diff
@@ -1,42 +1,68 @@
 import { AdministratorRoles } from "../administration/types";
 import { GroupMinimal, Permissions } from "../groups/types";
 
+/* Business to consumer specific details */
 type UserB2C = {
+    /* The display name (optional) */
     display_name?: string;
+    /* The family name (optional) */
     family_name?: string;
+    /* The given name (optional) */
     given_name?: string;
+    /* The object ID */
     oid: string;
 };
 
-export type User = {
-    id: string;
+/* A user with the essential information */
+export type UserNested = {
+    /* Indicates if the user is an administrator */
     administrator: boolean;
+    /* The unique identifier */
+    id: string;
+    /* The user's handle or username */
     handle: string;
+};
+
+/* A sample user */
+export type User = UserNested & {
+    /* Indicates if user is active */
     active: boolean;
+    /* The B2C user information (optional) */
     b2c?: UserB2C;
+    /* The user's display name (optional) */
```

Comment:
```suggestion
    /* Their display name (optional) */
```

... ect for other fields